docs/manual/cluster-admin/troubleshooting.md

```diff
@@ -149,3 +149,36 @@ Step 5. In the dev box container, start the internal storage service by `./paict
+
+If some completed jobs are lost, please first identify the problem, then solve it by following the steps below. We also recommend you to check this before you upgrade to `v1.6.0`.
+
+#### How to identify this issue
+
+First, stop all PAI services. On master node, run `sudo umount /mnt/paiInternal/storage`. After unmounting, if folder `/mnt/paiInternal/storage/pgdata` exists, the problem can be identified. If `/mnt/paiInternal/storage/pgdata` doesn't exist, you can igore this issue safely.
+
+#### How to solve the problem
+
+If the problem happens, there will be two pieces of database data. Please first backup them.
+
+The backup process:
+
+1. Stop all PAI services. On master node, run `sudo umount /mnt/paiInternal/storage`. Please backup `/mnt/paiInternal/storage/pgdata`.
+2. In dev box container, run `./paictl.py service start -n cluster-configuration internal-storage`. Then go to master node, backup `/mnt/paiInternal/storage/pgdata` to a different places.
+
+We will merge the two database data, and then upgrade to `v1.6.0` or later version to resolve this issue. Please first install postgresql client 12 on master node. Then follow the steps below.
+
+1. In dev box container, stop all PAI services by `./paictl.py service stop`
+2. On master node, run `sudo umount /mnt/paiInternal/storage`.
+3. In dev box container, run `./paictl.py service start -n cluster-configuration postgresql`. Please notice the source code used in dev box container should be under `v1.6.0` in this step.
+4. On master node, connect to the database by `psql  -U root -h 127.0.0.1 -W openpai`. The default password is `rootpass`. Then create a table with name `frameworks_tmp` by `CREATE TABLE frameworks_tmp AS TABLE frameworks WITH NO DATA;`. And insert frameworks data into it: `INSERT INTO frameworks_tmp SELECT * FROM frameworks;`. Use `\q` to exit from the database terminal.
+5. Dump the database by `pg_dump -U root -h 127.0.0.1 -W  -t framework_events -t framework_history -t pod_events -t pods -t task_history -t tags -f exclude_frameworks_dump.sql -a openpai` and `pg_dump -U root -h 127.0.0.1 -W -t frameworks_tmp -f frameworks_tmp_dump.sql openpai`.
+6. In dev box container, run `./paictl.py service stop -n cluster-configuration postgresql`. Wait serveral minutes, run `./paictl.py service start -n cluster-configuration internal-storage postgresql`
+7. On master node, restore the data to `framework_tmp`: `psql  -U root -h 127.0.0.1 -W  -f frameworks_tmp_dump.sql openpai`
+8. Connect to the database `psql  -U root -h 127.0.0.1 -W openpai`. Then merge the tables by: `DELETE FROM frameworks WHERE name in ( SELECT name FROM frameworks_tmp );` and `INSERT INTO frameworks SELECT  * FROM frameworks_tmp;`.
```

Comment:
For this method, seems the data on host will replace the data on internal storage. Is it a safe way? If some status/events will be lost according to the method

Reply:
Events will not be lost. They are dumped to `exclude_frameworks_dump.sql` and then restored.

Reply:
Only frameworks with the same name will be overrided. These are usually running jobs thus the influence is minor.